tests/envoy-smoke-test.sh

```diff
@@ -0,0 +1,342 @@
+  SERVER_IP=$SERVER1_IP;
+  SERVER_IP4=$SERVER1_IP4;
+}
+
+function service_init {
+  log "beginning service init"
+  cilium service update --frontend "$SVC_IP4:80" --id 2233 \
+			--backends "$SERVER1_IP4:80" \
+			--backends "$SERVER2_IP4:80"
+
+  cilium service update --frontend "[$SVC_IP]:80" --id 2234 \
+			--backends "[$SERVER1_IP]:80" \
+			--backends "[$SERVER2_IP]:80"
+
+  SERVER_IP=$SVC_IP
+  SERVER_IP4=$SVC_IP4
+  log "finished service init"
+}
+
+function proxy_init {
+  log "beginning proxy_init"
+  create_cilium_docker_network
+
+  if [ -z `docker ps -q -f name=^/server1$` ] ; then
+      docker run -dt --net=$TEST_NET --name server1 -l $SERVER_LABEL cilium/demo-httpd
+  fi
+  if [ -z `docker ps -q -f name=^/server2$` ] ; then
+      docker run -dt --net=$TEST_NET --name server2 -l $SERVER_LABEL cilium/demo-httpd
+  fi
+  if [ -z `docker ps -q -f name=^/client$` ] ; then
+      docker run -dt --net=cilium --name client -l id.client tgraf/netperf
+  fi
+  
+  SERVER1_IP=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.GlobalIPv6Address }}' server1)
+  SERVER2_IP=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.GlobalIPv6Address }}' server2)
+  SERVER1_IP4=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.IPAddress }}' server1)
+  SERVER2_IP4=$(docker inspect --format '{{ .NetworkSettings.Networks.cilium.IPAddress }}' server2)
+
+  SERVER_IP=$SERVER1_IP;
+  SERVER_IP4=$SERVER1_IP4;
+
+  wait_for_docker_ipv6_addr server1
+  wait_for_docker_ipv6_addr server2
+  wait_for_docker_ipv6_addr client
+
+  log "waiting for endpoints to get identities"
+  while [ `cilium endpoint list -o jsonpath='{range [*]}{.status.identity.id}{" "}{.status.identity.labels}{"\n"}' | grep '^[1-9][0-9]* .*id.\(server\|client\)' | cut -d ' ' -f1 | sort | uniq | wc -l` -ne 2 ] ; do
+    log "waiting..."
+    sleep 1
+  done
+
+  monitor_start
+  log "finished proxy_init"
+}
+
+function policy_single_egress {
+  cilium policy delete --all
+  cat <<EOF | policy_import_and_wait -
+[{
+    "endpointSelector": {"matchLabels":{"id.server":""}},
+    "ingress": [{
+        "fromEndpoints": [
+	    {"matchLabels":{"reserved:host":""}},
+	    {"matchLabels":{"id.client":""}}
+	]
+    }]
+},{
+    "endpointSelector": {"matchLabels":{"id.client":""}},
+    "egress": [{
+	"toPorts": [{
+	    "ports": [{"port": "80", "protocol": "tcp"}],
+	    "rules": {
+                "HTTP": [{
+		    "method": "GET",
+		    "path": "/public"
+                }]
+	    }
+	}]
+    }]
+}]
+EOF
+}
+
+function policy_many_egress {
+  cilium policy delete --all
+  cat <<EOF | policy_import_and_wait -
+[{
+    "endpointSelector": {"matchLabels":{"id.server":""}},
+    "ingress": [{
+        "fromEndpoints": [
+	    {"matchLabels":{"reserved:host":""}},
+	    {"matchLabels":{"id.client":""}}
+	]
+    }]
+},{
+    "endpointSelector": {"matchLabels":{"id.client":""}},
+    "egress": [{
+        "fromEndpoints": [{"matchLabels":{"id.server":""}}],
```

Comment:
Should this be `toEndpoints` since this is an egress rule?

Reply:
Yes! Due to this error the policy has applied to all endpoints, but the test only tests to 'server' so the test did not fail.